web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -441,4 +465,97 @@ export class BarChartComponent implements AfterViewInit {
       .call(brushSelector)
       .call(brushSelector.move, [firstItemPosition, lastItemPosition]);
   }
+
+  /**
+   * This function creates a boxplot next to the focus bar chart with all selected data.
+   * It indicates the maximum, the minimium, the median, the first quantile and the third
+   * quantile of the selected data duartions.
+   *
+   * @param inputData: Data selected for the focus bar chart
+   */
+  private createBoxplot(inputData: Item[]): void {
+    // Remove all elements of from the previous boxplot,
+    if (!this.boxplot) { return; }
+    this.boxplot.selectAll('circle').remove();
+    this.boxplot.selectAll('line').remove();
+    this.boxplot.selectAll('text').remove();
+
+    const durationSelected = inputData.map((obj: Item) => obj.durationHours);
+    const durationSorted = durationSelected.sort(d3.ascending);
+    const durationMax = durationSorted[durationSorted.length - 1];
+    const durationMin = durationSorted[0];
+
+    const q1 = d3.quantile(durationSorted, 0.25);
+    if (!q1) { return; }
+    const median = d3.quantile(durationSorted, 0.5);
+    if (!median) { return; }
+    const q3 = d3.quantile(durationSorted, 0.75);
+    if (!q3) { return; }
+
+    const boxWidth = 30;
+    // If more than 100 Items are selected, set the circle radius to be 1.4; otherwise, set
+    // the radius to 2.5.
+    const pointRadius = (inputData.length > BarChartComponent.DEFAULT_MAX_LABELS) ? 1.4 : 2.5;
+    const jitterWidth = 20;
+    const center = this.width - 60;
+
+    // Add individual points with jitter.
+    this.boxplot
+      .selectAll('points')
+      .data(inputData)
+      .enter()
+      .append('circle')
+        .attr('cx', (d: Item) => (center - jitterWidth / 2 + Math.random() * jitterWidth))
+        .attr('cy', (d: Item) => this.yScaleFocus(d.durationHours))
+        .attr('r', pointRadius)
+        .style('fill', (d: Item) => BarChartComponent.STATE_TO_COLOR[d.state])
+        .style('fill-opacity', 0.45); // Show overlay among dataPoints.
+
+    // Add the rectangle for the boxplot.
+    this.boxplot
+      .append('rect')
+        .attr('x', center - boxWidth / 2)
+        .attr('y', this.yScaleFocus(q3))
+        .attr('height', (this.yScaleFocus(q1) - this.yScaleFocus(q3)))
+        .attr('width', boxWidth)
+        .attr('stroke', BarChartComponent.COLOR_DARK_GRAY)
+        .style('fill-opacity', 0);
+
+    // Add the median horizontal line to the boxplot.
+    this.boxplot
+      .append('line')
+        .attr('x1', center - boxWidth / 2)
+        .attr('x2', center + boxWidth / 2)
+        .attr('y1', this.yScaleFocus(median))
+        .attr('y2', this.yScaleFocus(median))
+        .attr('stroke', BarChartComponent.COLOR_DARK_GRAY)
+        .style('fill-opacity', 1);
+
+    // Add ticks for max, min, median, q1 and q3.
+    this.boxplot
+      .selectAll('ticks')
+      .data([durationMax, durationMin, median, q1, q3])
```

Comment:
store `[durationMax, durationMin, median, q1, q3]` separately since you use it later as well